samples/cellular/udp/Kconfig

```diff
@@ -14,6 +14,14 @@ config UDP_DATA_UPLOAD_FREQUENCY_SECONDS
 	int "How often data is transmitted to the server"
 	default 900
 
+config UDP_DATA_UPLOAD_FOREVER
+	bool "Enable that data is transmitted to the server forever"
+	default y
+
+config UDP_DATA_UPLOAD_ITERATIONS
+	int "NUmber of data transmissions to the server before shutdown" if !UDP_DATA_UPLOAD_FOREVER
+	default 5
```

Comment:
We shouldn't need two entries for this; one config for the number of iterations should be sufficient.
If set to `-1` let the loop run forever.

Reply:
I thought about the -1 idea before, but I don't like the fact that a different value has a completely different behavior. While these two options make sense just by reading their name, in the -1 case the user will have to find such behavior in the documentation instead.

Reply:
@MechanicV1 made me aware that the `download_client` uses `-1` already in a similar way https://github.com/nrfconnect/sdk-nrf/blob/8f8c2689151063428e0f52645762ebedaecf78ab/subsys/net/lib/download_client/Kconfig#L112 so I updated this sample to align to that.